diff --git a/recipes/duckdb/all/conandata.yml b/recipes/duckdb/all/conandata.yml
index 8b2ec0fe..31f4b191 100644
--- a/recipes/duckdb/all/conandata.yml
+++ b/recipes/duckdb/all/conandata.yml
@@ -1,79 +1,27 @@
 sources:
-  "1.1.3":
-    url: "https://github.com/duckdb/duckdb/archive/refs/tags/v1.1.3.tar.gz"
-    sha256: "2aea0af898ad753fee82b776fea1bf78ccbc9648986e7f7a87372df5e74cdb98"
-  "1.1.2":
-    url: "https://github.com/duckdb/duckdb/archive/refs/tags/v1.1.2.tar.gz"
-    sha256: "a3319a64c390ed0454c869b2e4fc0af2413cd49f55cd0f1400aaed9069cdbc4c"
-  "1.1.1":
-    url: "https://github.com/duckdb/duckdb/archive/refs/tags/v1.1.1.tar.gz"
-    sha256: "a764cef80287ccfd8555884d8facbe962154e7c747043c0842cd07873b4d6752"
-  "1.1.0":
-    url: "https://github.com/duckdb/duckdb/archive/refs/tags/v1.1.0.tar.gz"
-    sha256: "d9be2c6d3a5ebe2b3d33044fb2cb535bb0bd972a27ae38c4de5e1b4caa4bf68d"
-  "1.0.0":
-    url: "https://github.com/duckdb/duckdb/archive/refs/tags/v1.0.0.tar.gz"
-    sha256: "04e472e646f5cadd0a3f877a143610674b0d2bcf9f4102203ac3c3d02f1c5f26"
-  "0.10.2":
-    url: "https://github.com/duckdb/duckdb/archive/refs/tags/v0.10.2.tar.gz"
-    sha256: "662a0ba5c35d678ab6870db8f65ffa1c72e6096ad525a35b41b275139684cea6"
-  "0.10.1":
-    url: "https://github.com/duckdb/duckdb/archive/refs/tags/v0.10.1.tar.gz"
-    sha256: "83bd4944c070fd0bd287fbe62919fa887f35d7422ba0fa66e13d4ed098f3791a"
   "0.9.2":
     url: "https://github.com/duckdb/duckdb/archive/refs/tags/v0.9.2.tar.gz"
     sha256: "afff7bd925a98dc2af4039b8ab2159b0705cbf5e0ee05d97f7bb8dce5f880dc2"
+  "0.9.0":
+    url: "https://github.com/duckdb/duckdb/archive/refs/tags/v0.9.0.tar.gz"
+    sha256: "3dbf3326a831bf0797591572440e81a3d6d668f8e33a25ce04efae19afc3a23d"
+  "0.8.1":
+    url: "https://github.com/duckdb/duckdb/archive/refs/tags/v0.8.1.tar.gz"
+    sha256: "a0674f7e320dc7ebcf51990d7fc1c0e7f7b2c335c08f5953702b5285e6c30694"
+
 patches:
-  "1.1.3":
-    - patch_file: "patches/1.1.1-0001-fix-cmake.patch"
-      patch_description: "install static of shared library, add installation for odbc extention"
-      patch_type: "portability"
-    - patch_file: "patches/1.1.1-0002-msvc-bicobj.patch"
-      patch_description: "add /bigobj flag"
-      patch_type: "portability"
-  "1.1.2":
-    - patch_file: "patches/1.1.1-0001-fix-cmake.patch"
-      patch_description: "install static of shared library, add installation for odbc extention"
-      patch_type: "portability"
-    - patch_file: "patches/1.1.1-0002-msvc-bicobj.patch"
-      patch_description: "add /bigobj flag"
-      patch_type: "portability"
-  "1.1.1":
-    - patch_file: "patches/1.1.1-0001-fix-cmake.patch"
-      patch_description: "install static of shared library, add installation for odbc extention"
-      patch_type: "portability"
-    - patch_file: "patches/1.1.1-0002-msvc-bicobj.patch"
-      patch_description: "add /bigobj flag"
-      patch_type: "portability"
-  "1.1.0":
-    - patch_file: "patches/1.1.0-0001-fix-cmake.patch"
-      patch_description: "install static of shared library, add installation for odbc extention"
-      patch_type: "portability"
-    - patch_file: "patches/1.1.0-0002-msvc-bicobj.patch"
-      patch_description: "add /bigobj flag"
-      patch_type: "portability"
-  "1.0.0":
-    - patch_file: "patches/1.0.0-0001-fix-cmake.patch"
+  "0.9.2":
+    - patch_file: "patches/0.9.2-0001-fix-cmake.patch"
       patch_description: "install static of shared library, add installation for odbc extention"
       patch_type: "portability"
-    - patch_file: "patches/1.0.0-0002-msvc-bicobj.patch"
-      patch_description: "add /bigobj flag"
-      patch_type: "portability"
-  "0.10.2":
-    - patch_file: "patches/0.10.2-0001-fix-cmake.patch"
+  "0.9.0":
+    - patch_file: "patches/0.9.0-0001-fix-cmake.patch"
       patch_description: "install static of shared library, add installation for odbc extention"
       patch_type: "portability"
-    - patch_file: "patches/0.10.1-0002-msvc-bicobj.patch"
-      patch_description: "add /bigobj flag"
-      patch_type: "portability"
-  "0.10.1":
-    - patch_file: "patches/0.10.1-0001-fix-cmake.patch"
+  "0.8.1":
+    - patch_file: "patches/0.8.1-0001-fix-cmake.patch"
       patch_description: "install static of shared library, add installation for odbc extention"
       patch_type: "portability"
-    - patch_file: "patches/0.10.1-0002-msvc-bicobj.patch"
-      patch_description: "add /bigobj flag"
-      patch_type: "portability"
-  "0.9.2":
-    - patch_file: "patches/0.9.2-0001-fix-cmake.patch"
-      patch_description: "install static of shared library, add installation for odbc extention"
+    - patch_file: "patches/0.8.1-0002-fix-clang.patch"
+      patch_description: "fix clang compile error"
       patch_type: "portability"
diff --git a/recipes/duckdb/all/conanfile.py b/recipes/duckdb/all/conanfile.py
index bc6bf23a..bb764a9d 100644
--- a/recipes/duckdb/all/conanfile.py
+++ b/recipes/duckdb/all/conanfile.py
@@ -1,7 +1,7 @@
 from conan import ConanFile
 from conan.errors import ConanInvalidConfiguration
 from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rm, rmdir, replace_in_file
-from conan.tools.build import check_min_cppstd, cross_building
+from conan.tools.build import check_min_cppstd
 from conan.tools.scm import Version
 from conan.tools.microsoft import is_msvc
 from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
@@ -9,7 +9,7 @@ from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
 import glob
 import os

-required_conan_version = ">=2.1"
+required_conan_version = ">=1.53.0"

 class DuckdbConan(ConanFile):
     name = "duckdb"
@@ -25,6 +25,7 @@ class DuckdbConan(ConanFile):
         "fPIC": [True, False],
         "with_autocomplete": [True, False],
         "with_icu": [True, False],
+        "with_parquet": [True, False],
         "with_tpch": [True, False],
         "with_tpcds": [True, False],
         "with_fts": [True, False],
@@ -45,6 +46,7 @@ class DuckdbConan(ConanFile):
         "fPIC": True,
         "with_autocomplete": False,
         "with_icu": False,
+        "with_parquet": False,
         "with_tpch": False,
         "with_tpcds": False,
         "with_fts": False,
@@ -72,8 +74,8 @@ class DuckdbConan(ConanFile):
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
-        if Version(self.version) >= "1.1.0":
-            del self.options.with_odbc
+        if Version(self.version) >= "0.9.0":
+            del self.options.with_parquet

     def configure(self):
         if self.options.shared:
@@ -84,7 +86,7 @@ class DuckdbConan(ConanFile):

     def requirements(self):
         # FIXME: duckdb vendors a bunch of deps by modify the source code to have their own namespace
-        if self.options.get_safe("with_odbc"):
+        if self.options.with_odbc:
             self.requires("odbc/2.3.11")
         if self.options.with_httpfs:
             self.requires("openssl/[>=1.1 <4]")
@@ -106,35 +108,47 @@ class DuckdbConan(ConanFile):
         tc.variables["DUCKDB_MINOR_VERSION"] = Version(self.version).minor
         tc.variables["DUCKDB_PATCH_VERSION"] = Version(self.version).patch
         tc.variables["DUCKDB_DEV_ITERATION"] = 0
-        tc.variables["OVERRIDE_GIT_DESCRIBE"] = f"v{self.version}"
-
-        build_extensions = ""
-        if self.options.with_icu:
-            build_extensions += ";icu"
-        if self.options.with_autocomplete:
-            build_extensions += ";autocomplete"
-        if self.options.with_tpch:
-            build_extensions += ";tpch"
-        if self.options.with_tpcds:
-            build_extensions += ";tpcds"
-        if self.options.with_fts:
-            build_extensions += ";fts"
-        if self.options.with_visualizer:
-            build_extensions += ";visualizer"
-        if self.options.with_httpfs:
-            build_extensions += ";httpfs"
-        if self.options.with_json:
-            build_extensions += ";json"
-        if self.options.with_excel:
-            build_extensions += ";excel"
-        if self.options.with_inet:
-            build_extensions += ";inet"
-        if self.options.with_sqlsmith:
-            build_extensions += ";sqlsmith"
-        tc.variables["BUILD_EXTENSIONS"] = build_extensions
-
-        if "with_odbc" in self.options:
-            tc.variables["BUILD_ODBC_DRIVER"] = self.options.with_odbc
+
+        if "with_parquet" in self.options:
+            tc.variables["BUILD_PARQUET_EXTENSION"] = self.options.with_parquet
+
+        if Version(self.version) >= "0.9.0":
+            build_extensions = ""
+            if self.options.with_icu:
+                build_extensions += ";icu"
+            if self.options.with_autocomplete:
+                build_extensions += ";autocomplete"
+            if self.options.with_tpch:
+                build_extensions += ";tpch"
+            if self.options.with_tpcds:
+                build_extensions += ";tpcds"
+            if self.options.with_fts:
+                build_extensions += ";fts"
+            if self.options.with_visualizer:
+                build_extensions += ";visualizer"
+            if self.options.with_httpfs:
+                build_extensions += ";httpfs"
+            if self.options.with_json:
+                build_extensions += ";json"
+            if self.options.with_excel:
+                build_extensions += ";excel"
+            if self.options.with_inet:
+                build_extensions += ";inet"
+            if self.options.with_sqlsmith:
+                build_extensions += ";sqlsmith"
+            tc.variables["BUILD_EXTENSIONS"] = build_extensions
+        else:
+            tc.variables["BUILD_ICU_EXTENSION"] = self.options.with_icu
+            tc.variables["BUILD_TPCH_EXTENSION"] = self.options.with_tpch
+            tc.variables["BUILD_TPCDS_EXTENSION"] = self.options.with_tpcds
+            tc.variables["BUILD_FTS_EXTENSION"] = self.options.with_fts
+            tc.variables["BUILD_HTTPFS_EXTENSION"] = self.options.with_httpfs
+            tc.variables["BUILD_VISUALIZER_EXTENSION"] = self.options.with_visualizer
+            tc.variables["BUILD_JSON_EXTENSION"] = self.options.with_json
+            tc.variables["BUILD_EXCEL_EXTENSION"] = self.options.with_excel
+            tc.variables["BUILD_SQLSMITH_EXTENSION"] = self.options.with_sqlsmith
+
+        tc.variables["BUILD_ODBC_DRIVER"] = self.options.with_odbc
         tc.variables["FORCE_QUERY_LOG"] = self.options.with_query_log
         tc.variables["BUILD_SHELL"] = self.options.with_shell
         tc.variables["DISABLE_THREADS"] = not self.options.with_threads
@@ -145,8 +159,6 @@ class DuckdbConan(ConanFile):
         tc.variables["ENABLE_UBSAN"] = False
         if is_msvc(self) and not self.options.shared:
             tc.preprocessor_definitions["DUCKDB_API"] = ""
-        if Version(self.version) >= "0.10.0" and cross_building(self):
-            tc.variables["DUCKDB_EXPLICIT_PLATFORM"] = f"{self.settings.os}_{self.settings.arch}"
         tc.generate()

         dpes = CMakeDeps(self)
@@ -197,11 +209,8 @@ class DuckdbConan(ConanFile):
                 "duckdb_fastpforlib",
                 "duckdb_mbedtls",
             ]
-            self.cpp_info.libs.append("duckdb_fsst")
-            if Version(self.version) >= "0.10.0":
-                self.cpp_info.libs.append("duckdb_skiplistlib")
-            if Version(self.version) >= "0.10.3":
-                self.cpp_info.libs.append("duckdb_yyjson")
+            if Version(self.version) >= "0.6.0":
+                self.cpp_info.libs.append("duckdb_fsst")

             if self.options.with_autocomplete:
                 self.cpp_info.libs.append("autocomplete_extension")
@@ -219,8 +228,7 @@ class DuckdbConan(ConanFile):
                 self.cpp_info.libs.append("visualizer_extension")
             if self.options.with_httpfs:
                 self.cpp_info.libs.append("httpfs_extension")
-            if (self.settings.os == "Linux" and
-                (Version(self.version) < "0.10.1" or self.settings.arch == "x86_64")):
+            if Version(self.version) >= "0.6.0" and self.settings.os == "Linux":
                 self.cpp_info.libs.append("jemalloc_extension")
             if self.options.with_json:
                 self.cpp_info.libs.append("json_extension")
@@ -236,5 +244,8 @@ class DuckdbConan(ConanFile):

         if self.settings.os == "Windows":
             self.cpp_info.system_libs.append("ws2_32")
-            if Version(self.version) >= "0.10.0":
-                self.cpp_info.system_libs.extend(["rstrtmgr", "bcrypt"])
+
+        if self.options.with_shell:
+            binpath = os.path.join(self.package_folder, "bin")
+            self.output.info(f"Appending PATH env var: {binpath}")
+            self.env_info.PATH.append(binpath)
diff --git a/recipes/duckdb/all/patches/0.8.1-0001-fix-cmake.patch b/recipes/duckdb/all/patches/0.8.1-0001-fix-cmake.patch
new file mode 100644
index 000000000..5cd9417ef
--- /dev/null
+++ b/recipes/duckdb/all/patches/0.8.1-0001-fix-cmake.patch
@@ -0,0 +1,87 @@
+diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
+index 0200c15..008c20e 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -75,24 +75,26 @@ else()
+       duckdb_fastpforlib
+       duckdb_mbedtls)
+ 
++  if(BUILD_SHARED_LIBS)
+   add_library(duckdb SHARED ${ALL_OBJECT_FILES})
+   target_link_libraries(duckdb ${DUCKDB_LINK_LIBS})
+   link_threads(duckdb)
+   link_extension_libraries(duckdb)
+-
++  else()
+   add_library(duckdb_static STATIC ${ALL_OBJECT_FILES})
+   target_link_libraries(duckdb_static ${DUCKDB_LINK_LIBS})
+   link_threads(duckdb_static)
+   link_extension_libraries(duckdb_static)
+-
++  endif()
++  if(BUILD_SHARED_LIBS)
+   target_include_directories(
+     duckdb PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+                   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+-
++  else()
+   target_include_directories(
+     duckdb_static PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+                          $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+-
++  endif()
+   install(
+     DIRECTORY "${PROJECT_SOURCE_DIR}/src/include/duckdb"
+     DESTINATION "${INSTALL_INCLUDE_DIR}"
+@@ -104,9 +106,18 @@ else()
+ 
+ endif()
+ 
++if(BUILD_SHARED_LIBS)
+ install(
+-  TARGETS duckdb duckdb_static
++  TARGETS duckdb
+   EXPORT "${DUCKDB_EXPORT_SET}"
+   LIBRARY DESTINATION "${INSTALL_LIB_DIR}"
+   ARCHIVE DESTINATION "${INSTALL_LIB_DIR}"
+   RUNTIME DESTINATION "${INSTALL_BIN_DIR}")
++else()
++install(
++  TARGETS duckdb_static
++  EXPORT "${DUCKDB_EXPORT_SET}"
++  LIBRARY DESTINATION "${INSTALL_LIB_DIR}"
++  ARCHIVE DESTINATION "${INSTALL_LIB_DIR}"
++  RUNTIME DESTINATION "${INSTALL_BIN_DIR}")
++endif()
+diff --git a/tools/odbc/CMakeLists.txt b/tools/odbc/CMakeLists.txt
+index 8f13cfe..6755894 100644
+--- a/tools/odbc/CMakeLists.txt
++++ b/tools/odbc/CMakeLists.txt
+@@ -53,6 +53,14 @@ add_library(
+ set_target_properties(duckdb_odbc PROPERTIES DEFINE_SYMBOL "DUCKDB_ODBC_API")
+ target_link_libraries(duckdb_odbc ${LINK_LIB_LIST} duckdb_static)
+ 
++install(
++    TARGETS duckdb_odbc
++    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
++    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
++    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
++    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
++)
++
+ if(NOT WIN32 AND NOT CLANG_TIDY)
+   add_subdirectory(test)
+ endif()
+diff --git a/tools/sqlite3_api_wrapper/CMakeLists.txt b/tools/sqlite3_api_wrapper/CMakeLists.txt
+index 84ac539..f5597a4 100644
+--- a/tools/sqlite3_api_wrapper/CMakeLists.txt
++++ b/tools/sqlite3_api_wrapper/CMakeLists.txt
+@@ -26,7 +26,7 @@ if(NOT AMALGAMATION_BUILD)
+ endif()
+ link_threads(sqlite3_api_wrapper_static)
+ 
+-if(NOT WIN32)
++if(BUILD_SHARED_LIBS AND NOT WIN32)
+   add_library(sqlite3_api_wrapper SHARED ${SQLITE_API_WRAPPER_FILES})
+   target_link_libraries(sqlite3_api_wrapper duckdb ${DUCKDB_EXTRA_LINK_FLAGS})
+   link_threads(sqlite3_api_wrapper)
diff --git a/recipes/duckdb/all/patches/0.8.1-0002-fix-clang.patch b/recipes/duckdb/all/patches/0.8.1-0002-fix-clang.patch
new file mode 100644
index 00000000..227f7ea0
--- /dev/null
+++ b/recipes/duckdb/all/patches/0.8.1-0002-fix-clang.patch
@@ -0,0 +1,98 @@
+diff --git a/src/core_functions/aggregate/distributive/arg_min_max.cpp b/src/core_functions/aggregate/distributive/arg_min_max.cpp
+index 2fa9929158..35499878dc 100644
+--- a/src/core_functions/aggregate/distributive/arg_min_max.cpp
++++ b/src/core_functions/aggregate/distributive/arg_min_max.cpp
+@@ -133,8 +133,8 @@ struct ArgMinMaxBase {
+ 			return;
+ 		}
+ 		if (!target.is_initialized || COMPARATOR::Operation(source.value, target.value)) {
+-			STATE::template AssignValue(target.arg, source.arg, target.is_initialized);
+-			STATE::template AssignValue(target.value, source.value, target.is_initialized);
++			STATE::AssignValue(target.arg, source.arg, target.is_initialized);
++			STATE::AssignValue(target.value, source.value, target.is_initialized);
+ 			target.is_initialized = true;
+ 		}
+ 	}
+@@ -144,7 +144,7 @@ struct ArgMinMaxBase {
+ 		if (!state.is_initialized) {
+ 			finalize_data.ReturnNull();
+ 		} else {
+-			STATE::template ReadValue(finalize_data.result, state.arg, target);
++			STATE::ReadValue(finalize_data.result, state.arg, target);
+ 		}
+ 	}
+
+@@ -209,7 +209,7 @@ struct VectorArgMinMaxBase : ArgMinMaxBase<COMPARATOR> {
+ 			return;
+ 		}
+ 		if (!target.is_initialized || COMPARATOR::Operation(source.value, target.value)) {
+-			STATE::template AssignValue(target.value, source.value, target.is_initialized);
++			STATE::AssignValue(target.value, source.value, target.is_initialized);
+ 			AssignVector(target, *source.arg, 0);
+ 			target.is_initialized = true;
+ 		}
+diff --git a/src/core_functions/aggregate/distributive/bitagg.cpp b/src/core_functions/aggregate/distributive/bitagg.cpp
+index 52b671d96f..136e410f2a 100644
+--- a/src/core_functions/aggregate/distributive/bitagg.cpp
++++ b/src/core_functions/aggregate/distributive/bitagg.cpp
+@@ -50,10 +50,10 @@ struct BitwiseOperation {
+ 	template <class INPUT_TYPE, class STATE, class OP>
+ 	static void Operation(STATE &state, const INPUT_TYPE &input, AggregateUnaryInput &) {
+ 		if (!state.is_set) {
+-			OP::template Assign(state, input);
++			OP::Assign(state, input);
+ 			state.is_set = true;
+ 		} else {
+-			OP::template Execute(state, input);
++			OP::Execute(state, input);
+ 		}
+ 	}
+
+@@ -76,10 +76,10 @@ struct BitwiseOperation {
+ 		}
+ 		if (!target.is_set) {
+ 			// target is NULL, use source value directly.
+-			OP::template Assign(target, source.value);
++			OP::Assign(target, source.value);
+ 			target.is_set = true;
+ 		} else {
+-			OP::template Execute(target, source.value);
++			OP::Execute(target, source.value);
+ 		}
+ 	}
+
+diff --git a/src/core_functions/aggregate/distributive/minmax.cpp b/src/core_functions/aggregate/distributive/minmax.cpp
+index fa087bab38..6f2e34e286 100644
+--- a/src/core_functions/aggregate/distributive/minmax.cpp
++++ b/src/core_functions/aggregate/distributive/minmax.cpp
+@@ -428,7 +428,7 @@ struct VectorMinMaxBase {
+ 			if (!state.value) {
+ 				Assign(state, input, i);
+ 			} else {
+-				OP::template Execute(state, input, i, count);
++				OP::Execute(state, input, i, count);
+ 			}
+ 		}
+ 	}
+@@ -440,7 +440,7 @@ struct VectorMinMaxBase {
+ 		} else if (!target.value) {
+ 			Assign(target, *source.value, 0);
+ 		} else {
+-			OP::template Execute(target, *source.value, 0, 1);
++			OP::Execute(target, *source.value, 0, 1);
+ 		}
+ 	}
+
+diff --git a/src/execution/operator/aggregate/physical_window.cpp b/src/execution/operator/aggregate/physical_window.cpp
+index 7b879edf5b..84737e9436 100644
+--- a/src/execution/operator/aggregate/physical_window.cpp
++++ b/src/execution/operator/aggregate/physical_window.cpp
+@@ -384,7 +384,7 @@ private:
+ template <typename T, typename OP>
+ struct OperationCompare : public std::function<bool(T, T)> {
+ 	inline bool operator()(const T &lhs, const T &val) const {
+-		return OP::template Operation(lhs, val);
++		return OP::Operation(lhs, val);
+ 	}
+ };
+
diff --git a/recipes/duckdb/config.yml b/recipes/duckdb/config.yml
index edd51719..2eb48353 100644
--- a/recipes/duckdb/config.yml
+++ b/recipes/duckdb/config.yml
@@ -1,17 +1,8 @@
 versions:
-  "1.1.3":
-    folder: "all"
-  "1.1.2":
-    folder: "all"
-  "1.1.1":
-    folder: "all"
-  "1.1.0":
-    folder: "all"
-  "1.0.0":
-    folder: "all"
-  "0.10.2":
+  "0.9.2":
     folder: "all"
-  "0.10.1":
+  "0.9.0":
     folder: "all"
-  "0.9.2":
+  "0.8.1":
     folder: "all"
+
