diff --git a/recipes/m4/all/conanfile.py b/recipes/m4/all/conanfile.py
index dadcf2b1..b629f13c 100644
--- a/recipes/m4/all/conanfile.py
+++ b/recipes/m4/all/conanfile.py
@@ -49,6 +49,8 @@ class M4Conan(ConanFile):
         env.generate()
 
         tc = AutotoolsToolchain(self)
+        if self.settings.compiler in ["gcc", "clang"]:
+            tc.extra_cflags.append("-std=gnu17")
         if is_msvc(self):
             tc.extra_cflags.append("-FS")
             # Avoid a `Assertion Failed Dialog Box` during configure with build_type=Debug
